### PR TITLE
Update py-csvkit

### PR DESCRIPTION
**Pull request by glennpj**
This PR updates the py-csvkit package. This version requires a python
stack based on agate and this PR includes the new dependency packages.

- py-agate-dbf
- py-agate-excel
- py-agate-sql
- py-agate
- py-dbfread
- py-isodate
- py-leather
- py-parsedatetime
- py-python-slugify
- py-pytimeparse
- py-text-unidecode

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-agate-dbf/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class PyAgateDbf(PythonPackage):
+    """agate-dbf adds read support for dbf files to agate."""
+
+    homepage = "https://agate-dbf.readthedocs.io/en/latest/"
+    url      = "https://pypi.io/packages/source/a/agate-dbf/agate-dbf-0.2.1.tar.gz"
+
+    version('0.2.1', sha256='00c93c498ec9a04cc587bf63dd7340e67e2541f0df4c9a7259d7cb3dd4ce372f')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-agate',      type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate-dbf/package.py]:
```suggestion
    depends_on('py-agate@1.5.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate-dbf/package.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    depends_on('py-dbfread',    type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate-dbf/package.py]:
```suggestion
    depends_on('py-dbfread@2.0.5:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate-excel/package.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    """agate-excel adds read support for Excel files (xls and xlsx) to
+    agate."""
+
+    homepage = "https://agate-excel.readthedocs.io/en/latest/"
+    url      = "https://pypi.io/packages/source/a/agate-excel/agate-excel-0.2.3.tar.gz"
+
+    version('0.2.3', sha256='8f255ef2c87c436b7132049e1dd86c8e08bf82d8c773aea86f3069b461a17d52')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-agate',      type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate-excel/package.py]:
```suggestion
    depends_on('py-agate@1.5.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate-excel/package.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    depends_on('py-xlrd',       type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate-excel/package.py]:
```suggestion
    depends_on('py-xlrd@0.9.4:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate-excel/package.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    depends_on('py-openpyxl',   type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate-excel/package.py]:
```suggestion
    depends_on('py-openpyxl@2.3.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate-sql/package.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+class PyAgateSql(PythonPackage):
+    """agate-sql adds SQL read/write support to agate."""
+
+    homepage = "https://agate-sql.readthedocs.io/en/latest/"
+    url      = "https://pypi.io/packages/source/a/agate-sql/agate-sql-0.5.4.tar.gz"
+
+    version('0.5.4', sha256='9277490ba8b8e7c747a9ae3671f52fe486784b48d4a14e78ca197fb0e36f281b')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-agate',      type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate-sql/package.py]:
```suggestion
    depends_on('py-agate@1.5.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate-sql/package.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    depends_on('py-sqlalchemy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate-sql/package.py]:
```suggestion
    depends_on('py-sqlalchemy@1.0.8:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate/package.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    instead of machines. It is an alternative to numpy and pandas that solves
+    real-world problems with readable code."""
+
+    homepage = "https://agate.readthedocs.io/en/latest/"
+    url      = "https://pypi.io/packages/source/a/agate/agate-1.6.1.tar.gz"
+
+    version('1.6.1', sha256='c93aaa500b439d71e4a5cf088d0006d2ce2c76f1950960c8843114e5f361dfd3')
+
+    depends_on('py-setuptools',     type='build')
+    depends_on('py-six',            type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate/package.py]:
```suggestion
    depends_on('py-six@1.9.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate/package.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    depends_on('py-pytimeparse',    type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate/package.py]:
```suggestion
    depends_on('py-pytimeparse@1.1.5:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate/package.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    depends_on('py-parsedatetime',  type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate/package.py]:
```suggestion
    depends_on('py-parsedatetime@2.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate/package.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+    depends_on('py-babel',          type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate/package.py]:
```suggestion
    depends_on('py-babel@2.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate/package.py [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    depends_on('py-isodate',        type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate/package.py]:
```suggestion
    depends_on('py-isodate@0.5.4:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate/package.py [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    depends_on('py-python-slugify', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate/package.py]:
```suggestion
    depends_on('py-python-slugify@1.2.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-agate/package.py [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+    depends_on('py-leather',        type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-agate/package.py]:
```suggestion
    depends_on('py-leather@0.3.2:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-csvkit/package.py [hunk 15]
```diff
@@ -13,13 +13,17 @@ class PyCsvkit(PythonPackage):
     homepage = 'http://csvkit.rtfd.org/'
     url      = "https://pypi.io/packages/source/c/csvkit/csvkit-0.9.1.tar.gz"
 
+    version('1.0.4', sha256='1353a383531bee191820edfb88418c13dfe1cdfa9dd3dc46f431c05cd2a260a0')
     version('0.9.1', sha256='92f8b8647becb5cb1dccb3af92a13a4e85702d42ba465ce8447881fb38c9f93a')
 
-    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools',          type=('build', 'run'))
+    depends_on('py-six',                 type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-csvkit/package.py]:
```suggestion
    depends_on('py-six@1.6.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-csvkit/package.py [hunk 16]
```diff
@@ -13,13 +13,17 @@ class PyCsvkit(PythonPackage):
...
-    depends_on('py-dbf@0.94.003', type=('build', 'run'))
-    depends_on('py-xlrd', type=('build', 'run'))
-    depends_on('py-sqlalchemy', type=('build', 'run'))
-    depends_on('py-six', type=('build', 'run'))
-    depends_on('py-openpyxl@2.2.0-b1', type=('build', 'run'))
+    depends_on('py-dbf@0.94.003',        type=('build', 'run'), when='@0.9.1')
+    depends_on('py-xlrd',                type=('build', 'run'), when='@0.9.1')
+    depends_on('py-sqlalchemy',          type=('build', 'run'), when='@0.9.1')
+    depends_on('py-openpyxl@2.2.0-b1',   type=('build', 'run'), when='@0.9.1')
+    depends_on('py-agate',               type=('build', 'run'), when='@1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-csvkit/package.py]:
```suggestion
    depends_on('py-agate@1.6.1:', type=('build', 'run'), when='@1:')
```

File: var/spack/repos/builtin/packages/py-csvkit/package.py [hunk 17]
```diff
@@ -13,13 +13,17 @@ class PyCsvkit(PythonPackage):
...
+    depends_on('py-agate-excel',         type=('build', 'run'), when='@1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-csvkit/package.py]:
```suggestion
    depends_on('py-agate-excel@0.2.2:', type=('build', 'run'), when='@1:')
```

File: var/spack/repos/builtin/packages/py-csvkit/package.py [hunk 18]
```diff
@@ -13,13 +13,17 @@ class PyCsvkit(PythonPackage):
...
+    depends_on('py-agate-dbf',           type=('build', 'run'), when='@1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-csvkit/package.py]:
```suggestion
    depends_on('py-agate-dbf@0.2.0:', type=('build', 'run'), when='@1:')
```

File: var/spack/repos/builtin/packages/py-csvkit/package.py [hunk 19]
```diff
@@ -13,13 +13,17 @@ class PyCsvkit(PythonPackage):
...
+    depends_on('py-agate-sql',           type=('build', 'run'), when='@1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-csvkit/package.py]:
```suggestion
    depends_on('py-agate-sql@0.5.3:', type=('build', 'run'), when='@1:')
```

File: var/spack/repos/builtin/packages/py-csvkit/package.py [hunk 20]
```diff
@@ -13,13 +13,17 @@ class PyCsvkit(PythonPackage):
     homepage = 'http://csvkit.rtfd.org/'
     url      = "https://pypi.io/packages/source/c/csvkit/csvkit-0.9.1.tar.gz"
 
+    version('1.0.4', sha256='1353a383531bee191820edfb88418c13dfe1cdfa9dd3dc46f431c05cd2a260a0')
     version('0.9.1', sha256='92f8b8647becb5cb1dccb3af92a13a4e85702d42ba465ce8447881fb38c9f93a')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-csvkit/package.py]: Missing deps:
```python
depends_on('py-argparse@1.2.1:', type=('build', 'run'), when='^python@:2.6')
depends_on('py-ordereddict@1.1:', type=('build', 'run'), when='^python@:2.6')
depends_on('py-simplejson@3.6.3:', type=('build', 'run'), when='^python@:2.6')
```

File: var/spack/repos/builtin/packages/py-isodate/package.py [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+    valid as ISO date, because it is not explicitly forbidden.) Another example
+    is, when no time zone information is given for a time, then it should be
+    interpreted as local time, and not UTC.
+
+    As this module maps ISO 8601 dates/times to standard Python data types,
+    like date, time, datetime and timedelta, it is not possible to convert all
+    possible ISO 8601 dates/times. For instance, dates before 0001-01-01 are
+    not allowed by the Python date and datetime classes. Additionally
+    fractional seconds are limited to microseconds. That means if the parser
+    finds for instance nanoseconds it will round it to microseconds."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-isodate/package.py]: Probably don't need such a long description, one or two lines should suffice.

File: var/spack/repos/builtin/packages/py-leather/package.py [hunk 22]
```diff
@@ -0,0 +1,19 @@
...
+    """Leather is the Python charting library for those who need charts now and
+    don't care if they're perfect."""
+
+    homepage = "https://leather.readthedocs.io/en/stable/"
+    url      = "https://pypi.io/packages/source/l/leather/leather-0.3.3.tar.gz"
+
+    version('0.3.3', sha256='076d1603b5281488285718ce1a5ce78cf1027fe1e76adf9c548caf83c519b988')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-six',        type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-leather/package.py]:
```suggestion
    depends_on('py-six@1.6.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-python-slugify/package.py [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+class PyPythonSlugify(PythonPackage):
+    """A Python Slugify application that handles Unicode"""
+
+    homepage = "https://github.com/un33k/python-slugify"
+    url      = "https://pypi.io/packages/source/p/python-slugify/python-slugify-4.0.0.tar.gz"
+
+    version('4.0.0', sha256='a8fc3433821140e8f409a9831d13ae5deccd0b033d4744d94b31fea141bdd84c')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-text-unidecode', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-python-slugify/package.py]:
```suggestion
    depends_on('py-text-unidecode@1.3:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-python-slugify/package.py [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class PyPythonSlugify(PythonPackage):
+    """A Python Slugify application that handles Unicode"""
+
+    homepage = "https://github.com/un33k/python-slugify"
+    url      = "https://pypi.io/packages/source/p/python-slugify/python-slugify-4.0.0.tar.gz"
+
+    version('4.0.0', sha256='a8fc3433821140e8f409a9831d13ae5deccd0b033d4744d94b31fea141bdd84c')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-python-slugify/package.py]:
```python
depends_on('python@2.7:2.8,3.5:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-csvkit/package.py (hunk 20 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-csvkit/package.py]: I did not realize we were still supporting that. I will add those.

Comment (glennpj) [reply var/spack/repos/builtin/packages/py-csvkit/package.py]: These were added with latest commit.

File: var/spack/repos/builtin/packages/py-python-slugify/package.py (hunk 24 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-python-slugify/package.py]: I completely missed this one but it was added with latest commit.

**Comment by glennpj** [comment]
@adamjstewart I pushed a commit to address your changes. I am not sure why I did not add the version constraints on the dependencies. It is not like I did not see them. I will make sure to catch those in the future.

Thanks.

File: var/spack/repos/builtin/packages/py-isodate/package.py (hunk 21 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-isodate/package.py]: I shortened it.

**Pull request merged**